picosat.c

```diff
@@ -3903,43 +3905,40 @@ prop2 (PS * ps, Lit * this)
 
 #ifdef NO_BINARY_CLAUSES
   lstk = LIT2IMPLS (this);
-  if (lstk->count)
+  start = lstk->start;
+  l = start ? start + lstk->count : start;
```

Comment:
What is the purpose of this change? It seems to make the logic flow less clear to no obvious benefit.

Reply:
It checks for null before the UB `start + lstk->count`, and no-ops the subsequent while loop.

Reply:
...but if `lstk->count` is non-zero, the original code would SEGV, yes? Therefore, `!start` ⇒ `!count`. Since the old code cuts out the `while` loop, which is functionally equivalent to short-circuiting it, both versions appear to have the same effect?

Ultimately, it looks like a number of these changes result from an independent, pre-#1 attempt to fix things. But since #1 was merged, if two fixes are functionally equivalent, I think it would be simpler to keep the changes that are already merged, in order to reduce further churn and simplify review. (Applies here, at L4716, and L2819-2824.)

Reply:
Reverted my change in favor of text from PR#1.